pysr/regressor_sequence.py

```diff
@@ -0,0 +1,425 @@
+        return self._regressor.refresh(checkpoint_file=checkpoint_file)
+
+    def sympy(self, index=None):
+        return self._regressor.sympy(index=index)
+
+    def latex(self, index=None, precision=3):
+        return self._regressor.latex(index=index, precision=precision)
+
+    def get_hof(self):
+        return self._regressor.get_hof()
+
+    def latex_table(
+        self,
+        indices=None,
+        precision=3,
+        columns=["equation", "complexity", "loss", "score"],
+    ):
+        """Create a LaTeX/booktabs table for all, or some, of the equations.
+
+        Parameters
+        ----------
+        indices : list[int] | list[list[int]]
+            If you wish to select a particular subset of equations from
+            `self.equations_`, give the row numbers here. By default,
+            all equations will be used. If there are multiple output
+            features, then pass a list of lists.
+        precision : int
+            The number of significant figures shown in the LaTeX
+            representations.
+            Default is `3`.
+        columns : list[str]
+            Which columns to include in the table.
+            Default is `["equation", "complexity", "loss", "score"]`.
+
+        Returns
+        -------
+        latex_table_str : str
+            A string that will render a table in LaTeX of the equations.
+        """
+        self._regressor.refresh()
+
+        if isinstance(self._regressor.equations_, list):
+            if indices is not None:
+                assert isinstance(indices, list)
+                assert isinstance(indices[0], list)
+                assert len(indices) == self._regressor.nout_
+            variable_names = self.variable_names
+            if variable_names is not None:
+                variable_names = [
+                    variable_name + "t_0" for variable_name in variable_names
+                ]
+            else:
+                variable_names = [f"x{i}t_0" for i in range(self.n_features)]
+            table_string = sympy2multilatextable(
+                self._regressor.equations_,
+                indices=indices,
+                precision=precision,
+                columns=columns,
+                output_variable_names=variable_names,
+            )
+        elif isinstance(self.equations_, pd.DataFrame):
+            if indices is not None:
+                assert isinstance(indices, list)
+                assert isinstance(indices[0], int)
+            if self.variable_names is not None:
+                assert len(self.variable_names) == 1
+                variable_name = self.variable_names[0] + "t_0"
+            else:
+                variable_name = "xt_0"
+            table_string = sympy2latextable(
+                self._regressor.equations_,
+                indices=indices,
+                precision=precision,
+                columns=columns,
+                output_variable_name=variable_name,
+            )
+        else:
```

Comment:
Is all of this code needed? Perhaps there's a way to simply rely on `_regressor.latex_table`? 

Basically, you should try to write as little code as possible where possible. Makes maintenance (for me) much easier.

Reply:
unfortunately not because of the variable names
but should I change up latex_table in PySRRegressor to accept a variable names parameter or just keep it like it is

Reply:
Oh why doesn’t passing the normal variable names work (i.e., having PySRRegressir store them automatically after running .fit?)

Reply:
actually im gonna do that